release-notes/10.0/preview/preview1/winforms.md

```diff
@@ -1,14 +1,93 @@
+- `public void SetDataAsJson<T>(string format, T data)`
+- `public bool TryGetData<T>(string format, out T data)`
+- `public bool TryGetData<T>(string format, System.Func<System.Reflection.Metadata.TypeName, System.Type> resolver, out T data)`
+
+## New Configuration Switch
+
+- `Windows.ClipboardDragDrop.EnableUnsafeBinaryFormatterSerialization`
+
+## Considerations for BinaryFormatter usage with the Clipboard
+
+### Scenarios that work without any added reference to BinaryFormatter
+
+Applications that use the new .NET Clipboard APIs for standard data types or explicitly use JSON serialization will work seamlessly without needing any reference to `BinaryFormatter`. For instance, the new APIs like `SetDataAsJson` and `TryGetData<T>` are designed to handle common .NET types such as primitive types and `System.Drawing.Bitmap` without relying on `BinaryFormatter`. These APIs ensure type safety and provide mechanisms to constrain incoming data, making them suitable for most scenarios where data exchange is straightforward and does not involve complex types. Serialization using these methods is considered best practice, and developers are strongly encouraged to move in this direction.
+
+### Scenarios that require full backwards compatibility in serialization
+
+For applications that still rely on `BinaryFormatter` for custom format serialization or need to maintain complete compatibility with .NET Framework applications, a reference to the out-of-band, unsupported `System.Runtime.Serialization.Formatters` package and the corresponding AppContext switch is required as detailed [here](https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/compatibility-package) . This is particularly necessary for scenarios where complex types are involved, and the application cannot migrate to the new JSON-based APIs. In such cases, developers must explicitly opt-in to enable BinaryFormatter serialization and deserialization by setting the `ClipboardDragDrop.EnableUnsafeBinaryFormatterSerialization` switch to `true` in the configuration file in addition to `EnableUnsafeBinaryFormatterSerialization` flag to `true`. The advantage to this approach is that `BinaryFormatter` can be explicitly disabled for Clipboard scenarios, while still being allowed in other parts of the application. This ensures that the application can continue to function while gradually transitioning to safer alternatives.
+
+### Best practices to consider when migrating to the new Clipboard APIs
+
+1. *Replace obsolete APIs*  Start by replacing the obsolete `GetData` and `SetData` methods with the new `TryGetData<T>` and `SetDataAsJson` methods. This ensures type safety and avoids the risks associated with unbounded deserialization.
+1. *Use JSON Serialization:* For complex types, use JSON serialization instead of BinaryFormatter. The new APIs like `SetDataAsJson` and `TryGetData<T>` are designed to handle common .NET types and JSON serialization.

```

Comment:
```suggestion
1. *Use JSON Serialization:* For complex types, use JSON serialization instead of BinaryFormatter, replace `SetData` with the new APIs like `SetDataAsJson`.  The `TryGetData<T>` APIs are designed to handle common .NET types and JSON serialization.

```